app/services/annotated_commodity_service.rb

```diff
@@ -10,6 +10,9 @@ def initialize(heading)
   end
 
   def call
+    # All commodities start as leaf commodities until we discover otherwise
+    @heading.commodities.each { |commodity| commodity.leaf = true }
```

Comment:
Moved this here since it was previously done outside of the AnnotatedCommodityService and is definitely a concern of it. Testing this by killing the leaf values on the file inputs for the tests of this service.